californium-core/src/main/java/org/eclipse/californium/core/network/stack/BlockwiseLayer.java

```diff
@@ -731,10 +790,16 @@ private void handleBlock1Response(final Exchange exchange, final Response respon
 
 				clearBlock1Status(key, status);
 
+			} else if (enableAutoFailoverOn413 && response.getCode() == ResponseCode.REQUEST_ENTITY_TOO_LARGE) {
+				if (status.getCurrentNum() == 0 && block1.getSize() < status.getCurrentSize()) {
+					sendBlock(exchange, response, key, status, 0, block1.getSzx());
+				}else {
+					clearBlock1Status(key, status);
+					deliverResponseToUpperLayer(exchange,response);
+				}
```

Comment:
I would prefer, if the failover is already handled so that this null case could be kept simple here.